src/commands/webViewCommand.ts

```diff
@@ -0,0 +1,156 @@
+import path = require("path");
+import * as vscode from "vscode";
+import { getCodebashingLink } from "../codebashing/codebashing";
+import { Logs } from "../models/logs";
+import { AstResult } from "../models/results";
+import { getLearnMore } from "../sast/learnMore";
+import { getChanges, triageSubmit } from "../utils/triage";
+import { applyScaFix } from "../sca/scaFix";
+import { commands } from "../utils/common/commands";
+import { AstDetailsDetached } from "../views/resultsView/astDetailsView";
+import { AstResultsProvider } from "../views/resultsView/astResultsProvider";
+
+export class WebViewCommand {
+  context: vscode.ExtensionContext;
+  logs: Logs;
+  detailsPanel: vscode.WebviewPanel | undefined;
+  resultsProvider: AstResultsProvider;
+  constructor(context: vscode.ExtensionContext, logs: Logs, resultsProvider: AstResultsProvider) {
+    this.context = context;
+    this.detailsPanel = undefined;
+    this.logs = logs;
+    this.resultsProvider = resultsProvider;
+  }
+
+  public registerNewDetails(): vscode.Disposable {
+    return vscode.commands.registerCommand(
+      commands.newDetails,
+      async (result: AstResult, type?: string) => {
+        const detailsDetachedView = new AstDetailsDetached(
```

Comment:
@tiagobcx  this is a long method, can we break it ?

Reply:
Done splited it in creating the Webview, loading the learn more, changes tab and handle the communication between js and the webview